torchtitan/experiments/gpt_oss/model/model.py

```diff
@@ -0,0 +1,480 @@
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import math
+from typing import Tuple
+
+import torch
+from torch import nn
+from torch.distributed.tensor import DTensor
+from torchtitan.models.attention import build_attention
+from torchtitan.protocols.train_spec import ModelProtocol
+
+from .args import GptOssModelArgs
+from .moe import MoE
+
+# TODO: may be able to remove this once parallelized properly
+def convert_submodules_to_bf16(
```

Comment:
What is this function for? Are you trying to apply mixed precision training by applying this function? Currently titan is handling it here: https://github.com/pytorch/torchtitan/blob/d240be0cf6793a5f09dc1bf718e56be346f4761a/torchtitan/distributed/utils.py#L217, and in train.py https://github.com/pytorch/torchtitan/blob/main/torchtitan/train.py#L460.